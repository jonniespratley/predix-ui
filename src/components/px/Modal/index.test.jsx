import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import PxModal from './';

describe('Modal', () => {
  test('should render', () => {
    const onClick = sinon.spy();
    const wrapper = shallow(<PxModal onBackdropClick={onClick}>Content</PxModal>);
    // wrapper.find('.px-overlay').simulate('click');
    //expect(onClick.calledOnce).toEqual(true);
    const tree = renderer.create(<PxModal
      headerText='Modal'
      acceptText='Confirm'
      reject='Cancel'
      />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
