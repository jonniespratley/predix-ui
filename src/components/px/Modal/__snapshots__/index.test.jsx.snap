// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render 1`] = `
<div
  className="llDqkv"
>
  <div
    className="px-overlay bwgVqy"
    onClick={[Function]}
    onKeyUp={[Function]}
  />
  <div
    className="gGaFPK"
    role="region"
  >
    <div>
      <h3
        className="SKTug"
      >
        Modal
      </h3>
      <div
        className="hhuebM"
      />
      <div
        className="flLJBN"
      >
        <button
          className="px-button cfGkUF"
          disabled={null}
          id="btnModalPositive"
          onClick={null}
          size={null}
          type={null}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
